applications/app/AppLoader.scala

```diff
@@ -16,7 +16,7 @@ import jobs.{SiteMapJob, SiteMapLifecycle}
 import model.{ApplicationContext, ApplicationIdentity}
 import services.ophan.SurgingContentAgentLifecycle
 import play.api.ApplicationLoader.Context
-import play.api.BuiltInComponentsFromContext
+import play.api.{BuiltInComponentsFromContext, OptionalDevContext}
```

Comment:
Is this needed? I assume it's covered by `FrontendApplicationLoader`?

Reply:
That was the fix coming from Adam ( https://github.com/guardian/frontend/pull/23450/commits/0732ff363320759b3c645df65d6cbc53383f1dc8 ). In a nutshell (copy pasting from @adamnfish  's DM)

The wire stuff uses macros to do magic, it works by trying to find an unambiguous value, with the desired type. Most of the values are already built-in via Play, but sometimes we need to do a little finagling to make sure a value of the correct type exists. (e.g. wrapping the DevContext in OptionalDevContext). 

That was required so that we could use the non-deprecated constructor for the class you asked me about so that's basically the answer to your question, but I fixed a few other things as well while I was there the error message you pasted said the constructor was deprecated, which is true so I looked at the non-deprecated constructors to find one we could use instead and this commit is the result of that journey.